test/helpers/name.go

```diff
@@ -17,14 +17,24 @@ limitations under the License.
 package helpers
 
 import (
+	"math/rand"
 	"strings"
 	"testing"
+	"time"
+	"unicode"
 )
 
 const (
+	letterBytes    = "abcdefghijklmnopqrstuvwxyz"
+	randSuffixLen  = 8
+	sep            = '-'
 	testNamePrefix = "Test"
 )
 
+func init() {
+	rand.Seed(time.Now().UTC().UnixNano())
```

Comment:
Please document why you explicitly need to seed the random number generator.

Reply:
Done.

Reply:
Please document "why", not "how". "How" is clear from the code. What's the big deal with a pseudo random generator for the "appendsuffix" method? Actually I would even prefer that so reproducing scenarios would be easier.

Reply:
The comment on the method `AppendRandomString` explained the reason:
```
// AppendRandomString will generate a random string that begins with prefix.
// This is useful if you want to make sure that your tests can run at the same
// time against the same environment without conflicting.
```
Or do you prefer moving the comment here?

Reply:
Initializing the random number generator is different from generating a random suffix. Please re-read my comment and reasoning.

Reply:
Sorry for making this back and forth. 
Suppose there is a scenario as below:
One user was running the tests with `go test ./test/e2e`, and one of the tests created a service called `ksvc-xyzabc` with the `AppendRandomString` function. 
But the user made some new changes, and wanted to test again before the previous test finished. So he/she run `go test ./test/e2e` again. 
Without this `Seed`, the `AppendRandomString` function will create the same name as the previous one, which is `ksvc-xyzabc`, and conflicts with the last test, since the generator will create the same random sequence.
So this `Seed` is added for avoid this scenario.

I also added more comment. Does it make sense to you?

Reply:
Thanks, now I understand the problem: if you don't seed the random number generator, rerunning a test will likely generate the same random names again, causing conflicts if the resources were not deleted.

I suggest clarifying your comment, though. Something like:
```
// Properly seed the random number generator so AppendRandomString() is actually random.
// Otherwise, rerunning tests will generate the same names afain, causing conflicts with
// already existing conflicts.
```

Also, I strongly suggest you dump the seed, so you can reproduce the name generation during debugging.

Reply:
Done.